From 10a269784fd425c2bef2c91f7c7a5ddd1a6d105b Mon Sep 17 00:00:00 2001
From: David Tardon <dtardon@redhat.com>
Date: Thu, 28 Aug 2014 16:10:11 +0200
Subject: [PATCH] cast arg to the right type

According to desc., typelib_TypeDescription can be safely used where
typelib_TypeDescriptionReference is expected.

Change-Id: Ibc3effd7e2894ba6ff738503e747f5157c146454
---
 bridges/source/cpp_uno/gcc3_linux_powerpc64/uno2cpp.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bridges/source/cpp_uno/gcc3_linux_powerpc64/uno2cpp.cxx b/bridges/source/cpp_uno/gcc3_linux_powerpc64/uno2cpp.cxx
index 5836d8b..b41cee6 100644
--- a/bridges/source/cpp_uno/gcc3_linux_powerpc64/uno2cpp.cxx
+++ b/bridges/source/cpp_uno/gcc3_linux_powerpc64/uno2cpp.cxx
@@ -233,7 +233,7 @@ static void callVirtualMethod(void * pThis, sal_uInt32 nVtableIndex,
                 "fmr    %0,     1\n\t"
                 : "=f" (dret), "=r" (r3), "=r" (r4) : );
 
-    MapReturn(r3, r4, dret, pReturnTypeDescr, pRegisterReturn);
+    MapReturn(r3, r4, dret, reinterpret_cast<typelib_TypeDescriptionReference *>(pReturnTypeDescr), pRegisterReturn);
 }
 
 // Macros for easier insertion of values to registers or stack
-- 
1.9.3

